bancard/gateway.py

```diff
@@ -333,11 +334,14 @@ def rollback(self, tx_id: int) -> Tuple[bool, dict]:
 
     def callback(self, data: dict):
         shop_process_id = data.get("operation").get("shop_process_id")
+        tx = Transaction.objects.get(id=shop_process_id)
         amount = data.get("operation").get("amount")
         currency = data.get("operation").get("currency")
         string = f"{self.priv_key}{shop_process_id}confirm{amount}{currency}"
         token = hashlib.md5(string.encode()).hexdigest()
-        if token != data.get("operation").get("token"):
+        if token != data.get("operation").get("token") and tx.token != data.get(
+            "operation"
+        ).get("token"):
```

Comment:
`tx.token` and `token` should have the same value. Why are you checking both?
Lines 338 to 341 create the token to validate that received data was not tampered with.
`token` variable should match both `tx.token` and `token` value inside data.